aviary/interface/graphical_input.py

```diff
@@ -1,63 +1,1104 @@
+
+        # Track changes to the canvas and frame width and sync them,
+        # also updating the scrollbar.
+        def _configure_interior(event):
+            # Update the scrollbars to match the size of the inner frame.
+            size = (interior.winfo_reqwidth(), interior.winfo_reqheight())
+            canvas.config(scrollregion="0 0 %s %s" % size)
+            if interior.winfo_reqwidth() != canvas.winfo_width():
+                # Update the canvas's width to fit the inner frame.
+                canvas.config(width=interior.winfo_reqwidth())
+        interior.bind('<Configure>', _configure_interior)
+
+        def _configure_canvas(event):
+            if interior.winfo_reqwidth() != canvas.winfo_width():
+                # Update the inner frame's width to fill the canvas.
+                canvas.itemconfigure(interior_id, width=canvas.winfo_width())
+        canvas.bind('<Configure>', _configure_canvas)
+
+class AviaryMissionEditor(tk.Tk):
+    def __init__(self):
+        super().__init__()
+        self.theme = "light"
+        self.pallete = {"light":{'background_primary':'#ffffff',
+                                 'foreground_primary':'#000000',
+                                 'foreground_secondary':'#999999',
+                                 'crosshair':'#EE0000',
+                                 'lines':['#0209c6','#aa00aa'],
+                                 'image':'dark_mode.png',
+                                 'hover':'#63ebeb'},
+                        "dark":{'background_primary':'#1e1e1e',
+                                'foreground_primary':'#FEFEFE',
+                                'foreground_secondary':'#CCCCCC',
+                                'crosshair':'#EE0000',
+                                'lines':['#00b6f2','#ffff00'],
+                                'image':'light_mode.png',
+                                'hover':'#007acc'}}
+        
+        self.style_combobox = ttk.Style()
+        self.style_combobox.theme_use("alt")
+
+        # updates image object inside pallete with PhotoImage object using absolute filepath
+        source_directory = os.path.abspath(os.path.dirname(__file__))
+        for theme_info in self.pallete.values():
+            theme_info["image"] = tk.PhotoImage(file = os.path.join(source_directory,theme_info["image"]))
+
+        self.title('Mission Design Utility')
+        self.protocol("WM_DELETE_WINDOW",self.close_window)
+        self.focus_set() # focus the window
+        # stores/retrieves persistant settings in source directory
+        self.persist_filename = os.path.join(source_directory,"persist_settings.json")
+
+        # ---------------------------------------------------------------
+        # window geometry definition, allows reuse of user's modified size/location
+        # tkinter size string format:  widthxheight+x+y ;  x,y are location
+        default_win_size = (900,500) # CHANGE THIS TO AFFECT DEFAULT SIZE
```

Comment:
Would it make sense or be helpful to have the window size be based on the user's actual monitor resolution? Could use logic like this: https://stackoverflow.com/questions/3129322/how-do-i-get-monitor-resolution-in-python/56913005#56913005

Reply:
I checked with different sizes and the minimize size has to be 800x500, otherwise the screen starts to hide information. However, I did set the default size to use either the minimum size or 50% of user's screen width/height, whichever size is larger. Is this functionality closer to what you had in mind? The user can also opt to store settings and in that case however they resize the window that's how it will open on the next run.